webviz_subsurface/plugins/_surface_with_seismic_cross_section.py

```diff
@@ -0,0 +1,502 @@
+                Input(self.ids("cube"), "value"),
+                Input(self.ids("surface"), "value"),
+                Input(self.ids("color-values"), "value"),
+                Input(self.ids("color-scale"), "colorscale"),
+            ],
+        )
+        def _render_fence(coords, cubepath, surfacepath, color_values, colorscale):
+            if not coords:
+                raise PreventUpdate
+            cube = load_cube_data(get_path(cubepath))
+            fence = get_fencespec(coords)
+            hmin, hmax, vmin, vmax, values = cube.get_randomline(fence)
+
+            surface = xtgeo.RegularSurface(str(get_path(surfacepath)))
+            s_arr = get_surface_fence(fence, surface)
+            return make_heatmap(
+                values,
+                s_arr=s_arr,
+                s_name=self.surfacenames[self.surfacefiles.index(surfacepath)],
+                colorscale=colorscale,
+                xmin=hmin,
+                xmax=hmax,
+                ymin=vmin,
+                ymax=vmax,
+                zmin=color_values[0],
+                zmax=color_values[1],
+                xaxis_title="Distance along polyline",
+                yaxis_title=self.zunit,
+            )
+
+        @app.callback(
+            [
+                Output(self.ids("color-values"), "min"),
+                Output(self.ids("color-values"), "max"),
+                Output(self.ids("color-values"), "value"),
+                Output(self.ids("color-values"), "step"),
+            ],
+            [Input(self.ids("color-range-btn"), "n_clicks")],
+            [State(self.ids("cube"), "value")],
+        )
+        def _update_color_slider(_clicks, cubepath):
+
+            cube = load_cube_data(get_path(cubepath))
+            minv = float(f"{cube.values.min():2f}")
+            maxv = float(f"{cube.values.max():2f}")
+            value = [minv, maxv]
+            step = (maxv - minv) / 100
+            return minv, maxv, value, step
+
+    def add_webvizstore(self):
+        return [
+            *[(get_path, [{"path": fn}]) for fn in self.segyfiles],
+            *[(get_path, [{"path": fn}]) for fn in self.surfacefiles],
+        ]
+
+
+# pylint: disable=too-many-arguments, too-many-locals
+def make_heatmap(
+    arr,
+    s_arr,
+    s_color="black",
+    s_name=None,
+    height=800,
+    zmin=None,
+    zmax=None,
+    xmin=None,
+    xmax=None,
+    ymin=None,
+    ymax=None,
+    colorscale=None,
+    uirevision=None,
+    showscale=True,
+    reverse_y=True,
+    text=None,
+    title=None,
+    yaxis_title=None,
+    xaxis_title=None,
+):
+
+    x_inc = (xmax - xmin) / arr.shape[1]
+    y_inc = (ymax - ymin) / arr.shape[0]
+    colors = (
+        [[i / (len(colorscale) - 1), color] for i, color in enumerate(colorscale)]
+        if colorscale
+        else "RdBu"
+    )
+    return {
+        "data": [
+            {
+                "type": "heatmap",
+                "name": "seismic",
+                "text": text if text else None,
```

Comment:
If wanted behavior is `text if text is not None else None`, we can replace that with only `text`.